### DISCUSSION TITLE
What is current practice with object-graph mapping?

### DISCUSSION THREAD

**Discussion by tmcsys** [ArcadeData/arcadedb#1397] [Q&A]
Where are we (Graph Database users) with OGM?  

Any suggestions for an OGM to use with ArcadeDB?

I am used to using JPA and Hibernate for relational db work.  It’s been a while since I looked at OGMs.

It looks like neo4j-ogm is currently in development, but Hibernate OGM has not been updated since 2018 and is no longer being maintained. 

Ferma was at one time actively developed as a Tinkertop layer and I experimented with it and OrientDB, but development of that project has stopped.  It looks like the Tinkertop interface has changed significantly so I see little chance that project returning to life.

**Comment by lvca** [comment]
The new trend is to avoid such a layer that gives a little automation but limits a lot as soon as you need something more than the base use cases. After all, using the ArcadeDB Java API could be the easiest way to use ArcadeDB and they will always supported 🙂
Unless you have hundreds of types, mapping a POJO into a vertex takes very little time
We could provide some basic auto mapping POJO <-> Vertex or Document to simplify this

**Comment by tmcsys** [comment]
It seems that Neo4j saw some value and put a lot of work into it?

**Comment by mdre** [comment]
Hi!
I`am finishing the port of ODBOGM to ADBOGM for Java development. 
It is a clear way to work with java object with Arcade. We are using it with OrientDB in a really large project since 2019 and now I`am trying to convert it to Arcade. I think in a near future we will move to Arcade. 
The OrientDB project is in 

https://github.com/mdre/odbogm

and the ArcadeDB work in progress is in 

https://github.com/mdre/adbogm (work in progress. Compile but fail due to a lot of lack of RemoteDatabase methods).
